src/components/layout/page-header/PageHeader.tsx

```diff
@@ -64,104 +105,138 @@ export interface PageHeaderProps {
+  const {
+    actions,
+    breadcrumbs,
+    description,
+    disableStickyActions = false,
+    hasDivider,
+    metadata,
+    navigation,
+    scrollBehavior = PageHeaderScrollBehavior.scroll,
+    title,
+    ...rest
+  } = props;
+
+  const [ref, setRef] = useForwardedRef(forwardedRef);
+  const { height = 0 } = useResizeObserver(ref);
+
+  // The actions are sticky when the PageHeader's behavior is `collapse`.
+  // `disableStickyActions` cancels that.
+  const stickyActions =
+    scrollBehavior === PageHeaderScrollBehavior.collapse && !disableStickyActions;
 
   return (
-    <StyledPageHeader ref={ref} {...rest}>
-      <StyledPageHeaderTop>
-        {breadcrumbs}
-
-        <StyledPageHeaderMain>
-          <StyledPageHeaderMainLeft>
-            {title}
-
-            {metadata}
-
-            {description}
-          </StyledPageHeaderMainLeft>
-
-          {actions && <StyledPageHeaderMainRight>{actions}</StyledPageHeaderMainRight>}
-        </StyledPageHeaderMain>
-      </StyledPageHeaderTop>
-
-      {(hasDivider || navigation) && (
-        <StyledPageHeaderBottom>
-          {navigation}
+    <StyledPageHeader
+      ref={setRef}
+      {...rest}
+      css={{
+        '--page-header-total-height': `${height}px`,
+        ...PageHeaderBehaviorStyles[scrollBehavior],
+      }}
+      hasFullWidthNav={!stickyActions}>
+      {breadcrumbs && <StyledPageHeaderBreadcrumbs>{breadcrumbs}</StyledPageHeaderBreadcrumbs>}
+
+      <StyledPageHeaderMain>
+        {title}
+
+        {metadata}
+
+        {description}
+      </StyledPageHeaderMain>
+
+      {actions && (
+        <StyledPageHeaderActions
+          style={
+            stickyActions
+              ? { top: cssVar('layout-page-header-actions-offset'), position: 'sticky' }
+              : {}
+          }>
+          {actions}
+        </StyledPageHeaderActions>
+      )}
 
-          {navigation ? <StyledDividerWithOverlap /> : <Divider />}
-        </StyledPageHeaderBottom>
+      {navigation && (
+        <>
+          <StyledPageHeaderBottom>{navigation}</StyledPageHeaderBottom>
+          <StyledDividerWithOverlap />
+        </>
       )}
+      {hasDivider && !navigation && <StyledDivider />}
     </StyledPageHeader>
   );
 });
 
 PageHeaderRoot.displayName = 'PageHeader';
 
-const StyledPageHeader = styled.div`
+const StyledPageHeader = styled.div<{ hasFullWidthNav: boolean }>`
   grid-area: ${PageGridArea.header};
 
-  display: flex;
-  align-items: flex-start;
-  flex-direction: column;
-  gap: ${cssVar('dimension-space-300')};
+  display: grid;
+
+  grid-template-columns: auto min-content;
+  grid-template-rows: repeat(4, auto);
+  grid-template-areas:
+    '${PageHeaderArea.breadcrumbs}  ${PageHeaderArea.breadcrumbs}'
+    '${PageHeaderArea.main}         ${PageHeaderArea.actions}'
+    '${PageHeaderArea.nav} ${({ hasFullWidthNav }) => (hasFullWidthNav ? PageHeaderArea.nav : '_')}'
```

Comment:
I wonder if this notation could be easier to understand by putting the condition around the whole grid-template-areas and repeating all the rest?

Reply:
yeah, I went back-and-forth on that pattern :smile: